pages/pipelines/security/oidc/aws.md

```diff
@@ -0,0 +1,205 @@
+
+    Only OIDC token exchange requests (for IAM roles) from Buildkite Agents with these IP addresses will be permitted.
+
+1. Verify that your custom trust policy is complete. The following example trust policy (noting that `AWS_ACCOUNT_ID` has not been specified) will only allow the exchange of an agent's OIDC tokens with IAM roles when:
+    * the Buildkite organization is `example-org`
+    * the Buildkite pipeline is `example-pipeline`
+    * building on both the `main` branch and tagged releases
+    * on Buildkite Agents whose IP addresses are either `192.0.2.0` or `198.51.100.0`
+
+    ```json
+    {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+                "Effect": "Allow",
+                "Principal": {
+                    "Federated": "arn\:aws\:iam:\:AWS_ACCOUNT_ID\:oidc-provider/agent.buildkite.com"
+                },
+                "Action": "sts:AssumeRoleWithWebIdentity",
+                "Condition": {
+                    "StringEquals": {
+                        "agent.buildkite.com:aud": "sts.amazonaws.com"
+                    },
+                    "StringLike": {
+                        "agent.buildkite.com:sub": [
+                            "organization\:example-org\:pipeline\:example-pipeline\:ref:refs/heads/main:*",
+                            "organization\:example-org\:pipeline\:example-pipeline\:ref:refs/tags/*:*"
+                        ]
+                    },
+                    "IpAddress": {
+                        "aws:SourceIp": [
+                            "192.0.2.0",
+                            "198.51.100.0"
+                        ]
+                    }
+
+                }
+            }
+        ]
+    }
+    ```
+
+1. In the **Custom trust policy** section, copy your modified custom trust policy, paste it into your IAM role, and complete the next few steps up to specifying the **Role name**.
+
+1. Specify an appropriate **Role name**, for example, `example-pipeline-oidc-for-ssm`, and complete the remaining steps.
+
+## Step 3: Configure your IAM role with AWS actions
+
+Add an inline or managed IAM policy to the role to allow the IAM role to perform any actions your pipeline needs. Common examples are permissions to read secrets from SSM and push images to ECR, but this entirely depends on the purpose of your Pipeline.
+
+In this example we’ll allow access to read an SSM Parameter Store key named `/pipelines/example-pipeline/oidc-for-ssm/example-deploy-key` by attaching the following inline policy.
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:GetParameters"
+            ],
+            "Resource": "arn\:aws\:ssm\:us-east-1\:012345678910\:parameter/pipelines/example-pipeline/oidc-for-ssm/example-deploy-key"
```

Comment:
@duckalini , just wanted to confirm that preceding the store key with `:parameter/` was correct?

Reply:
the preceding `:parameter/` is correct. 

This is a separate JSON file entirely and (basically) unrelated to the trust policy. 

The trust policy says **who** can assume the IAM role, this policy says what the IAM role can do once it's assumed. This is purely an example, and will likely be entirely different for each of our customers.